enhance: Add json key inverted index in stats for optimization

JsDove opened the pull request:
Add json key inverted index in stats for optimization

Commit:
add json key inverted index in stats to speetd up json expr

Signed-off-by: luzhang <luzhang@zilliz.com>
Co-authored-by: luzhang <luzhang@zilliz.com>

jsoncontainexpr unaryexpr binaryexpr json index optimization

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

improve jsonContainSExpr

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

add jsonindex test

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

support load json index after loadsegment

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

improve statschecker unittest

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

jsonindex expr code format

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix go format

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix controllerbasetest fail

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix jsonindex memeroy leak

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix jsonkey go format

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix jsonindex go codeformat

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

improve jsoninvert unitest

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

delete unuse code

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

refine test_json_key_index

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix cpp unitest

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

delete loginfo

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix complie error

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix codeformat

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix createindex again hang up

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix unitest

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix createindex hang

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix triggerstatstask go ut

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix jsonindex filter error

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix jsonindex filter error

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix format

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

improve jsonkey unitest

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

remove sealsegment chunknum assert

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix codereview

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix jsonindex parse string to int

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix unitest

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

improve

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix go test error

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

improve

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

adjust jsonkey tantivy buffer size

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix: nested jsonarray cause incorrect index

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix: nested jsonarray cause incorrect index

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

feat growingsegment jsonkey index

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

feat growingsegment jsonkey index

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix complie error

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix complie error

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix complie error

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix complie error

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix jsonkeyindex array contain

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix jsonkeyindex arrayindex filter error

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix jsonkeyindex arrayindex filter error

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

format

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

adjust createjsonindex trigger time

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

Revert "adjust createjsonindex trigger time"

This reverts commit 84ea08afeed19f91c0cc2d75ceabdf832ce8aabe.

fix jsoncontain

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix jsoncontain

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

growing jsonkey index no support offsetinput

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix yaml format

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

add jsonkeyindex unitest

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix e2e jsoncontain

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

bug fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

bug fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix go ut

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix ut

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

growing json key fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

test

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

test

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix test

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

test

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

test

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

test

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

test

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

remova skip

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

move upper to for {} outside

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

crashfix: concurrentvector json use view_element

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

rename variable

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

add json key stats enabled in yaml

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix complie error

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

modiy

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix jsonkey index read-write confict

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

revert jobmanager

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

revert

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

add growingjson index switch

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix codecheck

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix code check

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix lock

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix growing bitset out of size

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

remove mutx

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix gosdk test

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

replace interface

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

support strong query

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix: exists expr on nested dictionaries fail in JSONkeyIndex

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

compress jsonstatslog path only save filename

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

modify json key tantivy memory size

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix: Fix wrong trigger of json key index (#40026)

pr: https://github.com/milvus-io/milvus/pull/38039

Signed-off-by: zhenshan.cao <zhenshan.cao@zilliz.com>

fix: json key index gc

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix: json key index gc

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

Replace the position of EnabledJSONKeyStats

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

Set task version for stats task

Signed-off-by: Cai Zhang <cai.zhang@zilliz.com>

replace EnableJsonKeyStats

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

improve jsonkey gc logic

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix test

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix go test

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

compress jsonkey log

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix test

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

improve stats check logic

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix code review

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

polish name

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

polish name

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

polish

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

improve indexchecker

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

improve

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

improve stats check logic

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

add stats check test

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

Adjust the JSONIndexMemoryBudgetInTantivy  configuration position

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

polish name

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

improve stats check describecollection logic

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

json key stats filter element array with differenttype

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

replace JSONKeyStatsMemoryBudgetInTantivy unit from mb to byte

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix json stats gc

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

add jsonkeystats handoff

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix: [2.5]Replace the position of EnabledJSONKeyStats (#40108)

Replace the position of EnabledJSONKeyStats
issue: https://github.com/milvus-io/milvus/issues/36995
pr: https://github.com/milvus-io/milvus/pull/38039

---------

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

remove log

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix code format

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

improve json stats filter

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

filter

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

improve

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

improve

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

add jsonstats task limit

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

refine chunkervariablecolumn

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

refine chunkervariablecolumn

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

add test

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

add json key stats task limit

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix test case

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

polish name

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

add buildtype

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix go test

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix test

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

sort the map then add

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

polish name

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

add map

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

improve json stats filter

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

filter

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

improve

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

improve

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

add jsonstats task limit

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

refine chunkervariablecolumn

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

refine chunkervariablecolumn

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

add test

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

add json key stats task limit

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix test case

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

polish name

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

add buildtype

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix go test

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix test

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

sort the map then add

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

polish name

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

add map

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

add build json stats latency  metric

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix roll upgrade fail

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

polish name

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix rename

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix code format

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

add log

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

review fix upper jsondataformat

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix conflict

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix jsonstats term expr

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix negative number

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix negative number cannot filter out

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

JsDove commented:
rerun go-sdk

JsDove commented:
/run-cpu-e2e

JsDove commented:
rerun go-sdk

JsDove commented:
/run-cpu-e2e

JsDove commented:
rerun go-sdk

JsDove commented:
rerun go-sdk

JsDove commented:
/run-cpu-e2e

zhagnlu commented on internal/core/src/common/Json.h:
```diff
@@ -205,6 +278,62 @@ class Json {
+
+    template <typename T>
+    std::pair<T, std::string>
+    at_pos(uint16_t offset, uint16_t length) const {
```
still not used, remove

JsDove commented:
rerun go-sdk

JsDove commented:
rerun go-sdk

JsDove commented:
/run-cpu-e2e

JsDove commented:
rerun ut

JsDove commented:
/run-cpu-e2e

JsDove commented:
rerun go-sdk

JsDove commented:
/run-cpu-e2e

JsDove commented:
rerun go-sdk

JsDove commented:
/run-cpu-e2e

JsDove commented:
/run-cpu-e2e

JsDove commented:
rerun go-sdk

JsDove commented:
/run-cpu-e2e

JsDove commented:
rerun cpp-unit-test

JsDove commented:
rerun go-sdk

JsDove commented:
rerun go-sdk

JsDove commented:
rerun go-sdk

JsDove commented:
rerun cpp-unit-test

JsDove commented:
/run-cpu-e2e

JsDove commented:
/run-cpu-e2e

JsDove commented:
rerun go-sdk

JsDove commented:
rerun go-sdk

JsDove commented:
rerun go-sdk

zhagnlu commented on internal/core/src/index/JsonKeyInvertedIndex.cpp:
```diff
@@ -0,0 +1,245 @@
+    } else if (current.type == JSMN_STRING) {
+        Assert(current.size == 0);
+        AddInvertedRecord(
+            path, offset, current.start, current.end - current.start);
```
AddInvertedRecord(
            path, offset, current.start - 1, current.end - current.start + 1);  need add string ""

zhagnlu commented on internal/core/src/common/Json.h:
```diff
@@ -205,6 +238,21 @@ class Json {
+    at(uint16_t offset, uint16_t length) const {
+        if constexpr (std::is_same_v<T, std::string_view> ||
+                      std::is_same_v<T, std::string>) {
+            return value_result<T>(T(data_.data() + offset, length));
```
remove this convert, all using parse

JsDove commented:
rerun go-sdk

JsDove commented:
rerun go-sdk

JsDove commented:
/run-cpu-e2e

JsDove commented:
rerun go-sdk

JsDove commented:
rerun go-sdk

JsDove commented:
/run-cpu-e2e

JsDove commented:
rerun go-sdk

JsDove commented:
rerun go-sdk

JsDove commented:
rerun go-sdk

JsDove commented:
run ut

JsDove commented:
rerun ut

JsDove commented:
rerun go-sdk

JsDove commented:
/run-cpu-e2e

JsDove commented:
rerun go-sdk

JsDove commented:
rerun go-sdk

JsDove commented:
rerun cpp-unit-test

JsDove commented:
/run-cpu-e2e

JsDove commented:
/run-cpu-e2e

JsDove commented:
/run-cpu-e2e

JsDove commented:
/run-cpu-e2e

JsDove commented:
rerun go-sdk

JsDove commented:
/run-cpu-e2e

JsDove commented:
/run-cpu-e2e

JsDove commented:
run ut

JsDove commented:
rerun ut

JsDove commented:
/run-cpu-e2e

JsDove commented:
rerun go-sdk

JsDove commented:
rerun go-sdk

JsDove commented:
rerun go-sdk

JsDove commented:
rerun go-sdk

JsDove commented:
rerun ut

JsDove commented:
run ut

JsDove commented:
/run-cpu-e2e

zhagnlu commented on internal/datacoord/job_manager.go:
```diff
@@ -139,6 +141,22 @@ func needDoTextIndex(segment *SegmentInfo, fieldIDs []UniqueID) bool {
+		return false
+	}
+	for _, fieldID := range fieldIDs {
+		if segment.GetJsonKeyStats() == nil {
```
move upper to for {} outside

JsDove replied:
ok

JsDove commented:
rerun cpp-unit-test

JsDove commented:
rerun cpp-unit-test

zhagnlu commented on internal/core/src/index/JsonKeyInvertedIndex.cpp:
```diff
@@ -0,0 +1,335 @@
+            std::to_string(disk_file_manager_->GetFieldDataMeta().field_id);
+        d_type_ = TantivyDataType::Keyword;
+        wrapper_ = std::make_shared<TantivyIndexWrapper>(
+            field_name.c_str(), d_type_, path_.c_str(), false, false);
```
add JSON_INDEX_MEMORY_BUDGET param

zhagnlu commented on internal/core/src/exec/expression/ExistsExpr.cpp:
```diff
@@ -96,5 +100,45 @@ PhyExistsFilterExpr::EvalJsonExistsForDataSegment(OffsetVector* input) {
+            segment = dynamic_cast<const segcore::SegmentSealed*>(segment_);
+        }
+        auto field_id = expr_->column_.field_id_;
+        auto* index = segment->GetJsonKeyIndex(field_id);
+        Assert(index != nullptr);
+        auto filter_func = [segment, field_id, pointer](uint32_t row_id,
+                                                           uint16_t offset,
+                                                           uint16_t size) {
+            auto json_pair = segment->GetJsonData(field_id, row_id);
+            if (!json_pair.second) {
+                return false;
+            }
+            auto json =
+                milvus::Json(json_pair.first.data(), json_pair.first.size());
+            return json.exist(pointer);
+        };
+        cached_index_chunk_res_ =
+            index->FilterByPath(pointer, active_count_, filter_func).clone();
```
this logical is not correct, this expr target on json key, not value.

zhagnlu commented on internal/core/src/exec/expression/TermExpr.cpp:
```diff
@@ -76,6 +76,7 @@ PhyTermFilterExpr::Eval(EvalCtx& context, VectorPtr& result) {
                 break;
             }
             auto type = expr_->vals_[0].val_case();
+            auto start = std::chrono::steady_clock::now();
```
remove it

SpadeA-Tang commented on internal/core/thirdparty/tantivy/tantivy-binding/src/array.rs:
```diff
@@ -12,13 +12,13 @@ use crate::util::free_binding;
 #[repr(C)]
 pub struct RustArray {
-    array: *mut u32,
+    array: *mut i64,
```
why ranges to i64?

JsDove commented:
rerun ut

JsDove commented:
rerun ut

JsDove commented:
rerun go-sdk

JsDove commented:
rerun go-sdk

JsDove committed:
Fix the data is double  but the filtering is int

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

JsDove commented:
rerun ut

JsDove committed:
refactor jsonstats unary filter

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

JsDove committed:
fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

fix

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

JsDove committed:
fix indexchecker schema release

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

JsDove committed:
fix: jsonstats check if cache schema is nil lazy describecollection (#41068)

fix: jsonstats check if cache schema is nil lazy describecollection
pr:https://github.com/milvus-io/milvus/pull/38039
issue:https://github.com/milvus-io/milvus/issues/36995

---------

Signed-off-by: Xianhui.Lin <xianhui.lin@zilliz.com>

zhengbuqian commented on internal/core/src/mmap/ChunkedColumn.h:
```diff
@@ -147,6 +147,12 @@ class ChunkedColumnBase : public ColumnBase {
                   "GetBatchBuffer only supported for VariableColumn");
     }
 
+    virtual std::string_view
```
Why are we making this method virtual?

1. You updated `RawAt` of `ChunkedVariableColumn` to accept `size_t` instead of `int`, but did not mark `override` there. Luckily this won't cause a problem as `ChunkedVariableColumn` can still successfully override the method and polymorphism will continue to work.
2. `ChunkedArrayColumn` also has a `RawAt` method which is still accepting an `int` and returning a `ScalarArray`, this leads to method hiding issue and will lead to huge confusion:

```c++
ChunkedArrayColumn* array_column = ...;
array_column->RawAt(i); // calls ChunkedArrayColumn::RawAt and returns ScalarArray
ChunkedColumnBase* base = array_column;
base->RawAt(i); // calls ChunkedColumnBase::RawAt and returns string_view
```

This is not currently causing an issue because all `RawAt` calls on `ChunkedArrayColumn` object happens after the `ChunkedColumnBase` is carefully casted to `ChunkedArrayColumn`, but this can easily lead to hard to debug bugs.

zhengbuqian commented on internal/core/unittest/test_expr.cpp:
```diff
@@ -1381,6 +1401,272 @@ TEST_P(ExprTest, TestUnaryRangeJson) {
 
+    {
+        struct Testcase {
+            int64_t val;
```
should this be double? or is int64 intentional?
